tests/frameworks/test_starlette_middleware.py

```diff
@@ -0,0 +1,146 @@
+# (c) Copyright IBM Corp. 2021
+# (c) Copyright Instana Inc. 2020
+
+from typing import Generator
+
+import pytest
+from instana.singletons import agent, tracer
+from starlette.testclient import TestClient
+
+from tests.apps.starlette_app.app2 import starlette_server
+from tests.helpers import get_first_span_by_filter
+
+
+class TestStarletteMiddleware:
+    """
+    Tests Starlette with provided Middleware.
+    """
+
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
+        """SetUp and TearDown"""
+        # setup
+        # We are using the TestClient from Starlette to make it easier.
+        self.client = TestClient(starlette_server)
+        # Clear all spans before a test run.
+        self.recorder = tracer.span_processor
+        self.recorder.clear_spans()
+        yield
+        # teardown
+        # Clear all spans after a test run.
+        self.recorder.clear_spans()
```

Comment:
Can we remove this repetition?

```suggestion
        self.recorder.clear_spans()
```

Reply:
The idea was to make sure the span buffer is clean before the execution of a new test, but I will make the change.